File: client-app/src/app/models/customQuestion.ts
import { QuestionOption } from "./questionOption";
import { QuestionType } from "./questionType";

export interface CustomQuestion{
    id: string,
    registrationEventId: string,
    questionText: string,
    questionType: QuestionType
    required: boolean
    index: number
    options?: QuestionOption[];
}

File: Domain/CustomQuestion.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Domain
{
    public class CustomQuestion
    {
        public Guid Id { get; set; }
        public string QuestionText { get; set; }
        public QuestionType QuestionType { get; set; }
        public bool Required { get; set; }
        public int Index { get; set; }
        public Guid RegistrationEventId { get; set; }
        public RegistrationEvent RegistrationEvent { get; set; }

        public List<QuestionOption> Options { get; set; }

        public CustomQuestion()
        {
            Options = new List<QuestionOption>();
        }

    }
}
